staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -600,6 +600,8 @@ func (r *crdHandler) GetCustomResourceListerCollectionDeleter(crd *apiextensions
 	return info.storages[info.storageVersion].CustomResource, nil
 }
 
+var ExtraSerializers []func(runtime.ObjectCreater, runtime.ObjectTyper) runtime.SerializerInfo
```

Comment:
Global shared access to variables like this is something I'd like to avoid.

Reply:
At the very minimum, could we funnel access through a `SetXyzForTests` type function that clearly documents the limitations?  Curious if there are better ways to do this that anyone knows of (@deads2k and @sttts for any ideas)

Reply:
They should all be replaced by feature gate checks once the required integration tests are complete (https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/4222-cbor-serializer#phased-implementation). What do you think about introducing a separate `FeatureGate` object for this (one that is not wired to flags at runtime)?